internal/k8s/reconciler/gatewayclass.go

```diff
@@ -126,7 +126,7 @@ func (c *K8sGatewayClass) Validate(ctx context.Context) error {
 func (c *K8sGatewayClass) SyncStatus(ctx context.Context) error {
 	current := c.class.Status.Conditions
 	conditions := c.status.Conditions(c.class.Generation)
-	if !conditionsEqual(conditions, current) {
+	if !rstatus.ConditionsEqual(conditions, current) {
```

Comment:
This replaces the final usage of some old equality testing funcs that you'll now see removed below